pilot/pkg/model/destination_rule.go

```diff
@@ -80,37 +83,56 @@ func (ps *PushContext) mergeDestinationRule(p *consolidatedDestRules, destRuleCo
 	}
 
 	// DestinationRule does not exist for the resolved host so add it
-	p.destRule[resolvedHost] = &destRuleConfig
+	p.destRule[resolvedHost] = convertConsolidatedDestRule(&destRuleConfig)
 	p.exportTo[resolvedHost] = exportToMap
 }
 
 // inheritDestinationRule child config inherits settings from parent mesh/namespace
-func (ps *PushContext) inheritDestinationRule(parent, child *config.Config) *config.Config {
+func (ps *PushContext) inheritDestinationRule(parent, child *consolidatedDestRule) *consolidatedDestRule {
 	if parent == nil {
 		return child
 	}
 	if child == nil {
 		return parent
 	}
 
-	parentDR := parent.Spec.(*networking.DestinationRule)
+	if reflect.DeepEqual(parent.from, child.from) {
```

Comment:
can we use a more efficient check for equality?

Especially concerned that this is an ordered list - does order really matter here?

Reply:
https://github.com/istio/istio/pull/38088#discussion_r835743783 - that was my concern as well

Reply:
The order is determined, we have sorted all the drs before merge

Reply:
I can compare them one by one to improve efficiency

Reply:
But not sure how much it improves, since i can see many places in k8s use reflect.DeepEqual to compare labels.
I have never tested the two ways. Either make sense to me.

Reply:
improved